### DISCUSSION TITLE
Error Merubah Tipe Data Integer ke format Time

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#117] [Q&A] [answered]
Dear Team Algoritma,

Mohon bantuannya, saya mau merubah data dari format "int" menjadi bentuk "time" hasilnya error, code yang saya gunakan seperti ini:

ActvAcc_OtherBr_agg <- ActvAcc_OtherBr %>% 
  mutate(month_m = month(Maint_Date),
         time_maint = hms(Maint_Time, sec = 0, min = 1, hours = 2, roll = FALSE))

datanya di attachment. Saya coba gunakan fungsi format(), tetapi hasilnya data berubah menjadi tipe chr dan tidak ada pemisah waktunya.

Mohon petunjunya,
Thx u
 
<img width="422" alt="image" src="https://user-images.githubusercontent.com/103427989/170454394-00064069-ebce-4eb7-a22e-40d3a162bfaf.png">

**Comment by VictorNugraha** [comment, marked as answer]
Halo, @joejnt!

Disini saya ingin megkonfirmasi terlebih dahulu, apakah format data integer (kolom _**Maint_Time**_) yang ingin diubah dalam satuan detik?

**Reply by joejnt** [reply]
@VictorNugraha 

Betul, untuk data yg dimaksud di kolom Maint_Time

**Reply by VictorNugraha** [reply]
Halo, @joejnt!

Terima kasih atas konfirmasinya.

Fungsi yang mungkin bisa dicoba adalah fungsi **_seconds_to_period()_** dari **_library(lubridate)_**. Fungsi tersebut nantinya bisa di-isi dengan nama kolom yang ingin ditransformasi, untuk menjadikan data waktu dalam satuan detik menjadi bentuk period (...Day ...Hour ...Minute ...Second).

Untuk menggunakan fungsi tersebut Bapak bisa menerapkan-nya pada bagian fungsi **_mutate()_**, berikut contoh codenya:

```
df %>%
 mutate(nama_kolom_baru = seconds_to_period(Maint_Time))
```

Demikian informasi yang dapat saya sampaikan. Jika jawaban ini dapat membantu, mohon untuk meberikan tanda Answered pada jawaban ini. Selamat mencoba :D

**Reply by joejnt** [reply]
@VictorNugraha ,

well noted, terima kasih untuk insight nya.
